src/npctalk.cpp

```diff
@@ -2040,10 +2040,141 @@ void parse_tags( std::string &phrase, const Character &u, const Character &me, c
+                phrase.replace( fa, l, pgettext( "punctuation", "…" ) );
+                break;
+            case 2:
+                phrase.replace( fa, l, pgettext( "punctuation", "!" ) );
+                break;
+        }
+    } else if( tag.find( "<global_val:" ) == 0 ) {
+        //adding a global variable to the string
+        std::string var = tag.substr( tag.find( ':' ) + 1 );
+        // remove the trailing >
+        var.pop_back();
+        // resolve nest
+        recurse( var );
+        global_variables &globvars = get_globals();
+        phrase.replace( fa, l, globvars.get_global_value( "npctalk_var_" + var ) );
+    } else if( tag.find( "<item_name:" ) == 0 ) {
+        //embedding an items name in the string
+        std::string var = tag.substr( tag.find( ':' ) + 1 );
+        // remove the trailing >
+        var.pop_back();
+        // resolve nest
+        recurse( var );
+        // Try the variable as an item id
+        phrase.replace( fa, l, itype_id( var )->nname( 1 ) );
+    } else if( tag.find( "<item_description:" ) == 0 ) {
+        //embedding an items name in the string
+        std::string var = tag.substr( tag.find( ':' ) + 1 );
+        // remove the trailing >
+        var.pop_back();
+        // resolve nest
+        recurse( var );
+        // Try the variable as an item id
+        phrase.replace( fa, l, itype_id( var )->description.translated() );
+    } else if( tag.find( "<trait_name:" ) == 0 ) {
+        //embedding an items name in the string
+        std::string var = tag.substr( tag.find( ':' ) + 1 );
+        // remove the trailing >
+        var.pop_back();
+        // resolve nest
+        recurse( var );
+        // Try the variable as a trait id
+        phrase.replace( fa, l, trait_id( var )->name() );
+    } else if( tag.find( "<trait_description:" ) == 0 ) {
+        //embedding an items name in the string
+        std::string var = tag.substr( tag.find( ':' ) + 1 );
+        // remove the trailing >
+        var.pop_back();
+        // resolve nest
+        recurse( var );
+        // Try the variable as a trait id
+        phrase.replace( fa, l, trait_id( var )->desc() );
+    } else if( tag.find( "<spell_name:" ) == 0 ) {
+        //embedding an items name in the string
+        std::string var = tag.substr( tag.find( ':' ) + 1 );
+        // remove the trailing >
+        var.pop_back();
+        // resolve nest
+        recurse( var );
+        // Try the variable as a spell id
+        phrase.replace( fa, l, spell_id( var )->name.translated() );
+    } else if( tag.find( "<spell_description:" ) == 0 ) {
+        //embedding an items name in the string
+        std::string var = tag.substr( tag.find( ':' ) + 1 );
+        // remove the trailing >
+        var.pop_back();
+        // resolve nest
+        recurse( var );
+        // Try the variable as a spell id
+        phrase.replace( fa, l, spell_id( var )->description.translated() );
+    } else if( tag.find( "<city>" ) == 0 ) {
```

Comment:
What happens when a snippet containing these tags is expanded when a save is not loaded?

Reply:
`replace_city_tag` in `output.cpp` might conflict with this `<city>` flag.

Reply:
Good point. This should be left in the NPC version.